src/app/[locale]/requests/clearingRequest/detail/[id]/components/ClearingComments.tsx

```diff
@@ -10,24 +10,108 @@
 'use client'
 
 import { useTranslations } from 'next-intl'
-import { ClearingRequestDetails } from '@/object-types'
+import { Embedded, HttpStatus, ClearingRequestComments } from '@/object-types'
 import styles from '@/app/[locale]/requests/requestDetail.module.css'
 import { signOut, useSession } from 'next-auth/react'
+import { useCallback, useEffect, useState } from 'react'
+import { ApiUtils, CommonUtils } from '@/utils/index'
+import { notFound } from 'next/navigation'
+import MessageService from '@/services/message.service'
+import { Spinner } from 'react-bootstrap'
+import parse from 'html-react-parser'
+import Link from 'next/link'
 
 
-export default function ClearingComments({ data }: { data: ClearingRequestDetails }) {
+type EmbeddedClearingRequestComments = Embedded<ClearingRequestComments, 'sw360:comments'>
+
+export default function ClearingComments({ clearingRequestId }:
+                                         { clearingRequestId: string }) {
     const t = useTranslations('default')
-    const { status } = useSession()
+    const [loading, setLoading] = useState(true)
+    const { data:session, status } = useSession()
+    const [comments, setComments] = useState<Array<ClearingRequestComments>>([])
+    const [inputComment, setInputComment] = useState('')
+    const [commentPayload, setCommentPayload] = useState({
+        text: ''
+    })
+
+    const formatDate = (timestamp: number): string | null => {
```

Comment:
I think this function can be re-write as below
```
const formatDate = (timestamp: number): string | null => {
        if (!timestamp) {
            return null
        }
        const date = new Date(timestamp)
        const dateISOString = date.toISOString()
        return `${dateISOString.slice(0,10)}  ${dateISOString.slice(11,19)}`
    }
```

Reply:
@hoangnt2, I have refactored it.